libqtile/core/manager.py

```diff
@@ -288,9 +286,15 @@ def _process_screens(self) -> None:
             if not self.current_screen:
                 self.current_screen = scr
 
-            scr._configure(
-                self, i, x, y, w, h, self.groups[i],
-            )
+            if len(self.groups) < i + 1:
+                name = f"group_{i + 1}"
+                self.add_group(name)
```

Comment:
Now that I think about this more, this only solves the case when there is one more screen than groups. I think we should either add enough groups to match the number of screens or we should allow screens to be groupless.

Reply:
> this only solves the case when there is one more screen than groups

does it?

Reply:
...am I missing something? It doesn't iterate and only calls self.add_group once O.o

Reply:
nvm I am a dumb and apparently cannot read

Reply:
:D honestly i read it over a few times trying to figure out what mistake i had made lol